src/config/AccountConfig.ts

```diff
@@ -0,0 +1,29 @@
+import IAccountConfig from "../interfaces/Config/IAccountConfig";
+import {ALLOWED_ACCOUNT_CONFIG, INVALID_TYPE_MESSAGE} from "../constants";
+import Config from "./BaseConfig";
+
+class AccountConfig extends Config implements IAccountConfig {
+  provisioning_api_key?: string;
+  provisioning_api_secret?: string;
+  account_id?: string;
+  [other:string]: unknown;
+
+  /**
+   * @param {IAccountConfig} accountConfig
+   */
+  constructor(accountConfig: IAccountConfig | unknown) {
+    super();
+    if (this.isAccountConfigTypeGuard(accountConfig)) {
+      Object.assign(this, accountConfig);
+    } else {
+      console.error(INVALID_TYPE_MESSAGE);
+    }
+  }
+
+  isAccountConfigTypeGuard(accountConfig:unknown): accountConfig is IAccountConfig {
+    return this.isValidConfig(accountConfig, ALLOWED_ACCOUNT_CONFIG);
```

Comment:
Why not setting a member of the baseConfig with the ALLOWED values and let the baseConfig handle generically? 
Something like:
`this.validKeys = ALLOWED_ACCOUNT_CONFIG`

and in the `baseConfig` just use the `this.validKeys` in the `isValid` method